### PR TITLE
skip NaN in union and intersection

### PR DESCRIPTION
**Pull request by rafaqz**
Sister PR to https://github.com/JuliaGeo/GeoInterface.jl/pull/166

@asinghvi17

### CODE REVIEW

**Comment by rafaqz** [comment]
Hmm maybe this is wrong... if there is any NaN in an extent maybe the whole thing should be throw out? Currently this just throws out the individual NaN

**Comment by asinghvi17** [comment]
Hmm I'm not sure there.  It's not even that throwing out the extent is the safer option, since it means the spatial dimension is constrained to the non-NaN extent...

**Comment by rafaqz** [comment]
Yeah I think youre right, we just keep non-nan and thats as good as it can be, there are problems either way

**Comment by rafaqz** [comment]
Also is this breaking? I feel like its a pretty grey area, nan bounds are just undefined behaviour, and we can change undefined behaviour

**Comment by asinghvi17** [comment]
I wouldn't call this breaking for that reason

**Comment by rafaqz** [comment]
@evetion would be good to get your perspective on if this makes sense

**Comment by evetion** [comment]
I'm unsure about this, it's a grey area to be sure, and one might handle the spatial/non-spatial cases differently.

NaN spatially is akin to a missing coordinate. We have a skipmissing for that matter, so for calculating extents, skipping nan makes sense. However, if the nan ends up in an extent, it implies there was only a single coordinate (or none) in that dimensions. I'd say we error out, because there's no (or an unknown) extent in that dimension then.

Spatially, X and Y are intertwined, so an XY Extent that has a NaN is useless in the sense that most spatial operations (intersect, etc) can't work. If you're only interested in a single dimension, probably non-spatial, I guess one could handle it, but I would write down some use cases before implementing it, as most of the downstream behaviour is undefined.

Besides, in this PR the following snippet is unrealistic: `Y=(1.0, NaN)` no? How do you end up with a NaN for the max part and not also one for the min part? In that sense, replacing it by Y=(missing, missing) might almost be clearer.

TLDR: I don't think this makes sense, except for skipping nan numbers (akin to skipmissing) in calculating extents.

**Comment by rafaqz** [comment]
The thing is `union` is or can be used for calculating total extents do this is exactly that skipmissing case I'm thinking of! `intersectiom` is just for consistency.

`missing` was not an option as it's type unstable and that will propagate everywhere.

Currently NaN passed through and returns all NaN so erring really would be a breaking change

**Comment by asinghvi17** [comment]
bump @evetion

**Comment by evetion** [comment]
I think this should be closed. https://github.com/JuliaGeo/GeoInterface.jl/pull/166 should prevent Extents with NaN occuring, so this would not be needed.

I think (geospatial) set operations on NaN coordinates are undefined.
